.github/workflows/ci.yaml

```diff
@@ -0,0 +1,133 @@
+---
+name: "CI"
+on:
+  push:
+    branches: ["develop"]
+  pull_request:
+    branches: ["develop"]
+jobs:
+  Linting:
+    runs-on: "ubuntu-latest"
+    env:
+      INVOKE_NAUTOBOT_LOCAL: "True"
+    steps:
+      # ----------------------------------------------
+      #       check-out repo and set-up python
+      # ----------------------------------------------
+      - uses: "actions/checkout@v2"
+      - uses: "actions/setup-python@v2"
+        with:
+          python-version: 3.9
+      # ----------------------------------------------
+      #        load pip cache if cache exists
+      # ----------------------------------------------
+      - uses: "actions/cache@v2"
+        with:
+          path: "~/.cache/pip"
+          key: "${{ runner.os }}-pip"
+          restore-keys: "${{ runner.os }}-pip"
+      # ----------------------------------------------
+      #          install and run linters
+      # ----------------------------------------------
+      - run: "python -m pip install --upgrade pip wheel"
+      - run: "python -m pip install invoke black=='20.8b1' flake8=='3.9.2'"  # Until there is a poetry install --dev-only (poetry 1.2) install the linting tools here to speed up this step
+      - name: Setup Hadolint
+        run: |
+          mkdir -p $HOME/.local/bin
+          curl -Lo $HOME/.local/bin/hadolint https://github.com/hadolint/hadolint/releases/download/v2.0.0/hadolint-Linux-x86_64
+          chmod +x $HOME/.local/bin/hadolint
+          echo "$HOME/.local/bin" >> $GITHUB_PATH
+      - run: "invoke tests --lint-only"
+
+  Test:
+    needs: Linting
+    strategy:
+      fail-fast: true
+      matrix:
+        os: ["ubuntu-latest"]
+        python-version: [ "3.6", "3.7", "3.8", "3.9" ]
+    env:
+      INVOKE_NAUTOBOT_PYTHON_VER: ${{ matrix.python-version }}
+      INVOKE_NAUTOBOT_LOCAL: "True"
+      NAUTOBOT_SELENIUM_URL: http://localhost:4444/wd/hub 
+      NAUTOBOT_SELENIUM_HOST: 172.18.0.1
+      NAUTOBOT_DB_PASSWORD: decinablesprewad
+      NAUTOBOT_DB_USER: nautobot
+    services:
+      # Label used to access the service container
+      postgres:
+        # Docker Hub image
+        image: postgres:13
```

Comment:
Wondering if it would be good to have a matrix of postgres versions as well? At least minimum and latest supported versions?

Reply:
People are already throwing knives that the build takes too long!  lol we can add as much or as little as we want